core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+package org.apache.spark.status.protobuf;
+
+message JavaDate {
+  // The number of milliseconds since January 1, 1970, 00:00:00 GMT
+  optional int64 date = 1;
```

Comment:
shall we use int64 directly instead of having `JavaDate`? milliseconds since epoch is a very common definition and not tied to java.

Reply:
We will need utility methods for future encoding/decoding of java dates anyway. I prefer the current implementation. Looks more graceful.

Reply:
`JavaDate` is not really a data structure. Here it's more like a marker to indicate that the deserializer should turn it into Java `Date`. I think this should be the responsibility of `JobDataWrapperSerializer`, not the proto message.

Reply:
ok, changed to int64